### PR TITLE
fix(reth-evm-ethereum): no_std test compilation errors

### PR DESCRIPTION
**Pull request by martinezjorge**
Towards https://github.com/paradigmxyz/reth/issues/10088

In `reth-evm-ethereum`

The tests fail when running:

`cargo t -p reth-evm-ethereum --no-default-features`

These changes fix the compilation issues

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -321,7 +321,7 @@ reth-ethereum-forks = { path = "crates/ethereum-forks" }
 reth-ethereum-payload-builder = { path = "crates/ethereum/payload" }
 reth-etl = { path = "crates/etl" }
 reth-evm = { path = "crates/evm" }
-reth-evm-ethereum = { path = "crates/ethereum/evm" }
+reth-evm-ethereum = { path = "crates/ethereum/evm", default-features = false }
```
Comment (shekhirin) [inline Cargo.toml:324]: given that, we don't enable the `std` feature anywhere in the project. @mattsse we should propagate this change to all crates that use `reth-evm-ethereum` as a dependency, so they enable the `std` feature of it, right?

**Review by shekhirin — CHANGES_REQUESTED** [review]
It's a step in the right direction, and we will need this. This doesn't fully fix the compilation for `no_std`, because the crate still depends on `c-kzg`.

**Comment by martinezjorge** [comment]
@shekhirin

Thank you for the review. 

I'm happy to wait to see what @mattsse says in regards to setting `default-crates = false` on crates that depend on `reth-evm-ethereum`. 

I wasn't exactly aiming to progress #10088 in this PR but I'm happy to make it the focus. I was mostly aiming in this PR to just get the failing tests to pass when running it with the `--no-default-features` but maybe it affects things upstream of `reth-evm-ethereum` that I'm not aware of? It just seems to me that propagating `default-features = false` and `wasm-waspi1` support are separate tasks or is that not the case? Would love to hear your thoughts.

In regards to `wasm-waspi1` support, is the aim for the command:

`cargo +stable build -p reth-evm-ethereum --target wasm32-wasip1 --no-default-features`

to run successfully?

File: Cargo.toml (hunk 1 above)
Comment (martinezjorge) [reply Cargo.toml:324]: Maybe I'm missing something, but I'm not sure how `reth-evm-ethereum` depends on `c-kzg`. Could you please clarify that for me?

Comment (martinezjorge) [reply Cargo.toml:324]: The following dependencies in `reth-evm-ethereum` have `c-kzg` as default features:

1. `reth-primitives`
2. `reth-revm`

AFAIC, it seems that these are the main culprits; although, the rabbit hole may go deeper. Would the move be to go from `c-kzg` -> `kzg-rs` in the `reth-primitives` and `reth-revm`?

**Pull request closed without merging**